Modules/Quotas/src/Quotas.Application/DTOs/IndividualQuotaDTO.cs

```diff
@@ -16,4 +24,34 @@ public IndividualQuotaDTO(string id, MetricDTO metric, int max, QuotaPeriod peri
     public MetricDTO Metric { get; set; }
     public int Max { get; set; }
     public QuotaPeriod Period { get; set; }
+
+    public void CreateMappings(Profile configuration)
+    {
+        configuration
+            .CreateMap<IndividualQuota, IndividualQuotaDTO>()
+            .ForMember(dto => dto.Id,
+                expression => expression.MapFrom(quota => quota.Id.ToString()))
+            .ForMember(dto => dto.Metric,
+                expression => expression.MapFrom<MetricToMetricDtoResolver>())
+            .ForMember(dto => dto.Max,
+                expression => expression.MapFrom(quota => quota.Max))
+            .ForMember(dto => dto.Period,
+                expression => expression.MapFrom(quota => quota.Period));
+    }
+}
+
+public class MetricToMetricDtoResolver : IValueResolver<IndividualQuota, IndividualQuotaDTO, MetricDTO>
+{
+    private readonly IMetricsRepository _metricRepository;
+
+    public MetricToMetricDtoResolver(IMetricsRepository metricRepository)
+    {
+        _metricRepository = metricRepository;
+    }
+
+    public MetricDTO Resolve(IndividualQuota source, IndividualQuotaDTO destination, MetricDTO destMember, ResolutionContext context)
+    {
+        var metric = _metricRepository.FindAllWithKeys(new List<MetricKey> { source.MetricKey }, CancellationToken.None);
```

Comment:
`FindAllWithKeys` is an async method. Calling an async method in a non-async context is really bad in 99% of all cases, because it blocks the current thread. This can lead to really bad performance.